sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/FeatureManager.java

```diff
@@ -73,45 +91,178 @@ public Mono<Boolean> isEnabledAsync(String feature) {
+        return Flux.merge(results).reduce((a, b) -> a || b).single();
     }
 
-    private boolean isFeatureOn(FeatureFilterEvaluationContext filter, String feature) {
+    private Mono<Variant> generateVariant(String featureName, Object featureContext) {
+
+        VariantAssignment variantAssignment = new VariantAssignment(contextAccessor, evaluationOptions);
+
+        if (!StringUtils.hasText(featureName)) {
+            throw new IllegalArgumentException("Feature Variant name can not be empty or null.");
+        }
+
+        Feature feature = featureManagementConfigurations.getFeatureManagement().get(featureName);
+
+        if (feature == null) {
+            throw new FeatureManagementException("The Feature " + featureName + " can not be found.");
+        }
+
+        validateVariant(feature, featureName);
+
+        // Disabled?
+        if (!feature.getEvaluate() && StringUtils.hasText(feature.getAllocation().getDefautlWhenDisabled())) {
+            return variantAssignment.getVariant(feature.getVariants().values(),
+                feature.getAllocation().getDefautlWhenDisabled()).single();
+        } else if (!feature.getEvaluate()) {
+            return Mono.justOrEmpty(null);
+        } else if (feature.getEnabledFor().size() == 0) {
+            return variantAssignment.assignVariant(feature.getAllocation(), feature.getVariants().values());
+        }
+
+        List<Mono<Boolean>> results = new ArrayList<>();
+
+        for (FeatureFilterEvaluationContext featureFilter : feature.getEnabledFor().values()) {
+            if (StringUtils.hasText(featureFilter.getName())) {
+                results.add(isFeatureOn(featureFilter, feature.getKey(), featureContext));
+            }
+        }
+
+        Mono<Boolean> isEnabled;
+
+        // All Filters must be true
+        if (ALL_REQUIREMENT_TYPE.equals(feature.getRequirementType())) {
+            isEnabled = Flux.merge(results).reduce((a, b) -> a && b).single();
+        } else {
+            // Any Filter must be true
+            isEnabled = Flux.merge(results).reduce((a, b) -> a || b).single();
+        }
+
+        return isEnabled.flatMap(enabled -> {
+            if (!enabled && StringUtils.hasText(feature.getAllocation().getDefautlWhenDisabled())) {
+                return variantAssignment.getVariant(feature.getVariants().values(),
+                    feature.getAllocation().getDefautlWhenDisabled()).single();
+            } else if (!enabled) {
+                return Mono.justOrEmpty(null);
+            }
+
+            return variantAssignment.assignVariant(feature.getAllocation(), feature.getVariants().values());
+        });
+
+    }
+
+    private void validateVariant(Feature feature, String featureName) {
+        if (feature.getVariants() == null || feature.getVariants().size() == 0) {
+            throw new FeatureManagementException("The " + feature.getKey() + " has no assigned Variants.");
```

Comment:
What does `feature.getKey()` look like here? Seems like it should say `The feature` or `The key` to fit with the rest of the message.

Reply:
A feature key is also the feature flag name. So, the Beta flag has the key name Beta.

Reply:
Ok cool, yea then I think it makes sense to say "The feature"